lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -156,11 +170,24 @@ defmodule ExUnit.Assertions do
     match? = {:match?, meta, [left, Macro.var(:right, __MODULE__)]}
     pins = collect_pins_from_pattern(left, Macro.Env.vars(__CALLER__))
```

Comment:
Should the collection fo the pins & vars, as well as the expand_pattern be moved from the assertions module into the pattern module?